### ISSUE TITLE
Systematic localization

### ISSUE THREAD

**Issue by mayank99** [iTwin/stratakit#1059]
Many StrataKit components include built-in strings (typically for accessibility reasons). For example: [`Chip.DismissButton`](https://github.com/iTwin/design-system/blob/9635e0b55f7a380ed01573b23abd5a96ed008b8b/packages/structures/src/Chip.tsx#L128) and [`Dialog.CloseButton`](https://github.com/iTwin/design-system/blob/9635e0b55f7a380ed01573b23abd5a96ed008b8b/packages/structures/src/Dialog.tsx#L209).

These strings are currently not easy to localize. While it is technically possible, every instance needs to be painstakingly specified at the individual component level. When these strings are not visible on screen, it's very easy to miss localizing some of them, resulting in the page/app not being fully localized.

It would be nice to have a way to pass a "localization object" at the top level for localizing all things in  one go. This could be part of the `Root` component, or a separate `LocalizationProvider` component.

**Comment by mayank99** [comment]
This is no longer a concern because of https://github.com/iTwin/design-system/discussions/1116.

MUI has built-in localization: https://mui.com/material-ui/guides/localization/

See also: https://github.com/iTwin/stratakit/discussions/1485
